sra_experiment/nbconverted/DEG_validation.py

```diff
@@ -0,0 +1,311 @@
+# * Comparing the mean expression of PAO1 and PA14 we see that there are ~200 genes changed. This gives us some indication about what to expect for our DESeq analysis. However, we shouldn't expect the numbers to align because we are using different methods -- above we are comparing the raw gene expression values and looking for a threshold difference; below we DESeq fits the negative binomial model to the data and performs hypothesis testing to determine if there is a difference between the groups of samples.  
+
+# ### Differential expression analysis
+
+# In[19]:
+
+
+get_ipython().run_cell_magic('R', '', '# Select 59\n# Run one time\n#if (!requireNamespace("BiocManager", quietly = TRUE))\n#    install.packages("BiocManager")\n#BiocManager::install("DESeq2")')
+
+
+# In[20]:
+
+
+get_ipython().run_cell_magic('R', '', '# Load the DESeq2 library\nlibrary("DESeq2")')
+
+
+# In[21]:
+
+
+# Files to load into DE analysis (R)
+metadata_file = str(paths.SAMPLE_ANNOT_TEST)
+expression_data_file = str(paths.PAO1_GE_DE)
+out_file = str(paths.DE_STATS)
+
+
+# In[22]:
+
+
+# Check ordering of sample ids
+utils.check_sample_ordering(expression_data_file, metadata_file)
+
+
+# In[23]:
+
+
+get_ipython().run_cell_magic('R', '-i metadata_file -i expression_data_file -i out_file', "\nsource('../core_acc_modules/DE_analysis.R')\n\nget_DE_stats_DESeq(metadata_file,\n                   expression_data_file,\n                   out_file)")
+
+
+# In[24]:
+
+
+# Read in DE stats file
+DE_stats = pd.read_csv(paths.DE_STATS, sep='\t', header=0, index_col=0)
+print(DE_stats.shape)
+DE_stats.head()
+
+
+# In[25]:
+
+
+# Volcano plot
+DE_stats["-log10(padj)"] = -np.log10(DE_stats["padj"])
+sns.scatterplot(data=DE_stats, x="log2FoldChange", y="-log10(padj)")
+
+
+# ### Compare our DE results with publication
+
+# In[26]:
+
+
+# Get number of DEGs
+selected_DE_stats = DE_stats[(DE_stats['padj']<0.01)]
+
+print(selected_DE_stats.shape)
+selected_DE_stats.head()
+
+
+# In[27]:
+
+
+# Compare our findings against Sana et. al.
+degs_sana = ["PA3431", "PA3432", "PA1244", "PA4685"]
+for gene in degs_sana:
+    if gene in list(selected_DE_stats.index):
+        print(gene)
+
+
+# In[28]:
+
+
+# Compare genes whose mean gene expression differed between PAO1 and PA14
+# with those genes found using DESeq
+top_degs_by_padj = selected_DE_stats.index
+len(set(top_degs_by_padj).intersection(genes_found_from_GE))/len(genes_found_from_GE)
+
+
+# **Conclusions:**
+# 
+# Our DE analysis found ~1.1K significantly differentially expressed genes
+# 
+# (Check 1): The DEGs identified using DESeq (\~1.1K genes) is fairly consistent with the genes that were 1 standard deviation outside the correlation threshold (\~200 genes) -- there was a 75% overlap in these gene sets. This very roughly validates that DESeq is working as expected. I wouldn't expect the numbers to be this different though.
```

Comment:
This makes sense to me. DESeq is using a probability model that should (at least in theory) be a good match for the way RNA-seq data is generated, so I would expect it to have better statistical power to detect DE genes than your quick sanity check. The fact that what you found is (roughly) a subset of what DESeq found makes me fairly confident that this is working correctly.

Reply:
I agree...I was a little panicked because I thought they should still be closer in range, but they do appear to be a subset. I'll see if things change once i look more into tximport and DESeq and how they are processing the data (i.e. scaling between samples)